### PR TITLE
Added the user events to log entrance and exit

### PR DESCRIPTION
**Pull request by BrandonMurnane**

### CODE REVIEW

File: view/rooms/app.js [hunk 1]
```diff
@@ -520,6 +586,7 @@ function setupSocketIOClient(bots) {
   // Once connected to refocus display bots
   socket.on('connect', () => {
     debugMessage('Socket Connected');
+    userEnterRoom()
```
Comment (tausifmuzaffar) [inline view/rooms/app.js]: ;

File: view/rooms/app.js [hunk 2]
```diff
@@ -597,6 +664,8 @@ function setupSocketIOClient(bots) {
 
   socket.on('disconnect', () => {
     debugMessage('Socket Disconnected');
+    console.log('Socket Disconnected')
+    confirmUserExit()
```
Comment (tausifmuzaffar) [inline view/rooms/app.js]: ;

**Comment by tausifmuzaffar** [comment]
@BrandonMurnane update conflicts

File: view/rooms/app.js [hunk 3]
```diff
@@ -300,12 +300,15 @@ function createFooter(bot) {
...
   const linkedElement = document.createElement('a');
   const gitHubImage = document.createElement('img');
+  const botVersion = document.createElement('span');
 
   footer.className =
     'slds-section__title ' +
     'slds-p-horizontal_small ' +
     'slds-theme_shade ';
 
+  botVersion.innerHTML = 'Version '+bot.version;
```
Comment (iamigo) [inline view/rooms/app.js]: spaces around plus sign

Comment (iamigo) [reply view/rooms/app.js]: jscs does not ENFORCE style for everything under /view but you should run it anyway and bring your code up to style standards

File: view/rooms/app.js [hunk 4]
```diff
@@ -485,6 +489,68 @@ function createIframeEvent(channel, payload, bots, botId) {
...
+ *
+ * @returns {Promise} For use in chaining.
+ */
+function userEnterRoom() {
+  const currentUser = {
+    name: _user.name,
+    id: _user.id,
+    email: _user.email,
+    fullName: _user.fullName,
+  }
```
Comment (iamigo) [inline view/rooms/app.js]: style

File: view/rooms/app.js [hunk 5]
```diff
@@ -485,6 +489,68 @@ function createIframeEvent(channel, payload, bots, botId) {
...
+
+  const message = currentUser.fullName + ' has joined the room at ' +
+      moment().format('YYYY-MM-DD HH:mm Z');
+  const eventType =  {
+      'type': 'User',
+      'user': currentUser,
+      'isActive': true,
+    }
```
Comment (iamigo) [inline view/rooms/app.js]: style

Comment (kriscfoster) [reply view/rooms/app.js]: @BrandonMurnane Some users who did not sign in with SSO may not have a fullName field, we should probably fall back to name/email for this.

File: view/rooms/app.js [hunk 6]
```diff
@@ -485,6 +489,68 @@ function createIframeEvent(channel, payload, bots, botId) {
...
+    fullName: _user.fullName,
+  }
+
+  const eventType = {
+    'type': 'User',
+    'user': currentUser,
+    'isActive': false,
+  };
+
+  const message = currentUser.name + ' has left the room at ' +
```
Comment (kriscfoster) [inline view/rooms/app.js]: We should use fullName if there is one

**Comment by BrandonMurnane** [comment]
@tausifmuzaffar @iamigo @kriscfoster updated

**Pull request merged**